lib/prom_ex/plugins/plug_cowboy.ex

```diff
@@ -0,0 +1,185 @@
+        :plug_cowboy_http_event_metrics,
+        [
+          # Capture request duration information
+          distribution(
+            metric_prefix ++ [:http, :request, :duration, :milliseconds],
+            event_name: cowboy_stop_event,
+            measurement: :duration,
+            description: "The time it takes for the application to process HTTP requests.",
+            reporter_options: [
+              buckets: exponential!(1, 2, 12)
+            ],
+            drop: drop_ignored(ignore_routes),
+            tag_values: &get_tags/1,
+            tags: http_metrics_tags,
+            unit: {:native, :millisecond}
+          ),
+          distribution(
+            metric_prefix ++ [:http, :response, :duration, :milliseconds],
+            event_name: cowboy_stop_event,
+            measurement: :resp_duration,
+            description: "The time it takes for the application to send the HTTP response.",
+            reporter_options: [
+              buckets: exponential!(1, 2, 12)
+            ],
+            drop: drop_ignored(ignore_routes),
+            tag_values: &get_tags/1,
+            tags: http_metrics_tags,
+            unit: {:native, :millisecond}
+          ),
+          distribution(
+            metric_prefix ++ [:http, :request_body, :duration, :milliseconds],
+            event_name: cowboy_stop_event,
+            measurement: :req_body_duration,
+            description: "The time it takes for the application to receive the HTTP request body.",
+            reporter_options: [
+              buckets: exponential!(1, 2, 12)
+            ],
+            tag_values: &get_tags/1,
+            tags: http_metrics_tags,
+            unit: {:native, :millisecond}
+          ),
+
+          # Capture response payload size information
+          distribution(
+            metric_prefix ++ [:http, :response, :size, :bytes],
+            event_name: cowboy_stop_event,
+            measurement: :resp_body_length,
+            description: "The size of the HTTP response payload.",
+            reporter_options: [
+              buckets: exponential!(1, 4, 12)
+            ],
+            drop: drop_ignored(ignore_routes),
+            tag_values: &get_tags/1,
+            tags: http_metrics_tags,
+            unit: :byte
+          ),
+
+          # Capture the number of requests that have been serviced
+          counter(
+            metric_prefix ++ [:http, :requests, :total],
+            event_name: cowboy_stop_event,
+            description: "The number of requests have been serviced.",
+            drop: drop_ignored(ignore_routes),
+            tag_values: &get_tags/1,
+            tags: http_metrics_tags
+          )
+        ]
+      )
+    end
+
+    defp get_tags(%{resp_status: resp_status, req: %{method: method, path: path}}) do
+      case get_http_status(resp_status) do
+        status when is_binary(status) ->
+          %{
+            status: status,
+            method: method,
+            path: path
```

Comment:
I imagine that this tag can balloon out of control if it is not clamped. I.e it needs to be normalized so that routes like `/users/8bf0d5eb-5a9a-41ee-b7fe-c554d3593c00` are converted to `/users/:id`. The Phoenix plugin does this by looking up the route info via `Phoenix.Router.route_info`. Something similar will probably need to be done here.

Reply:
I could replicate the `Phoenix.Router.route_info` when Phx is available, but if not it's a little more tricky.

Do you know if there's something similar to `route_info` that can be used in a plug only (non phoenix) project 🤔